smdebug/pytorch/hook.py

```diff
@@ -69,6 +70,11 @@ def _get_num_workers(self):
                     return hvd.size()
             except (ModuleNotFoundError, ValueError, ImportError):
                 pass
+
+        if not is_first_process(self.out_dir):
+            self.save_all_workers = False
+            self.worker = "non_chief_worker"
```

Comment:
for all processes except the first get the name "non-chief_worker". will this conflict with the worker name assigned by get_worker_name() ?

Reply:
This overwrites the name assigned by `get_worker_name`